### PR TITLE
Cherry-picked webhook fix #712 for upcoming 0.9.1-rc.1

### PR DESCRIPTION
**Pull request by bwplotka**
Context: This release branch is blocked by https://github.com/GoogleCloudPlatform/prometheus-engine/issues/696. This PR is meant to cherry pick #712 which fixed this in the main branch. See more context about the fix in the #712

Signed-off-by: bwplotka <bwplotka@google.com>

### CODE REVIEW

**Comment by pintohutch** [comment]
Can you provide a little more context in the PR description for the approach and motivation?

**Comment by bwplotka** [comment]
Edited PR description PTAL

**Review by pintohutch — APPROVED** [review]

**Comment by pintohutch** [comment]
Approving if it unblocks. However, I'll defer to @bernot-dev for the domain expertise here.

**Review by bernot-dev — APPROVED** [review]

**Pull request merged**
